src/components/ButtonBase/ButtonBase.tsx

```diff
@@ -49,33 +49,33 @@ const ButtonBase = React.forwardRef<HTMLButtonElement, ButtonBaseProps>((props,
     onBlur,
     sx,
   } = props;
-  const color = '#123455';
   const testIdName = `${dataTestPrefixId}button`;
 
   return (
-    <button
-      ref={ref}
-      type={buttonType}
-      data-testid={generateTestDataId(testIdName, dataTestId)}
-      css={buttonBaseStyle({
-        isLoading,
-        isBlock,
-        color,
-        isDisabled,
-        iconLeft,
-        iconRight,
-        sx,
-      })}
-      onClick={(event) => {
-        if (onClick) {
-          onClick(event);
-        }
-      }}
-      onBlur={onBlur}
-      disabled={isDisabled || isLoading}
-    >
-      {children}
-    </button>
+    <div css={buttonWrapperStyle({ isBlock })}>
+      {isLoading && !isDisabled && <ButtonLoader type={type} dataTestId={testIdName} />}
+      <button
+        ref={ref}
+        type={buttonType}
```

Comment:
we changed the type to `buttonType` for consistency with HTML attributes so this could work -> https://www.w3schools.com/tags/att_button_type.asp